config/operator/role.yaml

```diff
@@ -4,6 +4,56 @@ kind: ClusterRole
 metadata:
   name: operator
 rules:
+- apiGroups:
+  - ""
+  resources:
+  - services
+  verbs:
+  - create
+  - delete
+  - get
+  - list
+  - patch
+  - update
+  - watch
+- apiGroups:
+  - apps
+  resources:
+  - deployments
+  verbs:
+  - create
+  - delete
+  - get
+  - list
+  - patch
+  - update
+  - watch
```

Comment:
I understand why this broad RBAC is added, but it effectively means the addon can do anything to all deployments in the cluster. Is there any chance the scaling sets are optional?

I can imagine a dual deployment workflow configured on the helm chart level
1) static - a single interceptor+scaler without scalingsets enabled (the current v0.8.0 workflow)
2) dynamic - enable scalingsets CRD and this broad RBAC

Reply:
^^ THIS